src/sdl/UserEvent.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+   Copyright (C) 2017 the Battle for Wesnoth Project http://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <SDL_events.h>
+
+class Wesnoth_SDL_UserEvent : public SDL_UserEvent
+{
+    public:
+        Wesnoth_SDL_UserEvent() {
+            memset (this, 0, sizeof(SDL_UserEvent));
```

Comment:
Why is there a space before the open parenthesis?

Reply:
(I wonder if you should do `static_cast<SDL_UserEvent*>(this)`... probably superfluous though?)

Reply:
Is there an stdlib function we can use here?

Reply:
The space is an error.

Why the static_cast?An attempt to access the base?

The use of memset is per the SOL docs. I might be able to use something else like any initislizer list. I'll look into it.

Reply:
Yeah, but I think the standard does mandate that the base class pointer is equal to the subclass pointer in this specific case, so it's probably superfluous.

@Vultraz - I think the closest thing the standard C++ library has to `memset` is `std:ininitialized_fill`; not sure if that's ppropriate here, though.